src/Microsoft.Health.SqlServer/Registration/SqlServerBaseRegistrationExtensions.cs

```diff
@@ -82,19 +81,18 @@ public static IServiceCollection AddSqlServerConnection(this IServiceCollection
              });
 
         // The following are only used in case of managed identity
-        services.AddSingleton<IAccessTokenHandler, ManagedIdentityAccessTokenHandler>();
-        services.AddSingleton<AzureServiceTokenProvider>(p =>
+        services.AddSingleton<IAccessTokenHandler, ManagedIdentityAccessTokenHandler>(p =>
         {
             SqlServerDataStoreConfiguration config = p.GetRequiredService<IOptions<SqlServerDataStoreConfiguration>>().Value;
 
-            string tokenProviderConnectionString = null;
+            string managedIdentityClientId = null;
 
             if (!string.IsNullOrEmpty(config.ManagedIdentityClientId))
             {
-                tokenProviderConnectionString = $"RunAs=App;AppId={config.ManagedIdentityClientId}";
+                managedIdentityClientId = config.ManagedIdentityClientId;
```

Comment:
I think we can just pass `config.ManagedIdentityClientId` directly to the ctor

Reply:
Or, simply update the ctor for `ManagedIdentityAccessTokenHandler` to use `IOptions<SqlServerDataStoreConfiguration>` to avoid this custom factory